tests/common/quantization/test_passes.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from pathlib import Path
+
+import pytest
+
+from nncf.quantization.passes import remove_dropout_nodes_inplace
+from tests.post_training.test_templates.models import NNCFGraphDropoutRemovingCase
+from tests.shared.nx_graph import compare_nx_graph_with_reference
+from tests.shared.paths import TEST_ROOT
+
+DATA_ROOT = TEST_ROOT / "common" / "data" / "reference_graphs"
+
+
+class TestModes(Enum):
+    VALID = "valid"
+    WRONG_TENSOR_SHAPE = "wrong_dropout_node"
+    WRONG_PARALLEL_EDGES = "wrong_parallel_edges"
+
+
+@pytest.mark.parametrize("mode", [TestModes.VALID, TestModes.WRONG_TENSOR_SHAPE, TestModes.WRONG_PARALLEL_EDGES])
+def test_remove_dropout_nodes_inplace(mode: TestModes):
```

Comment:
Why do we not add tests for each pass since we introduce this file in this PR?

Reply:
Because there was no tests for passes before, `get_inference_nncf_graph` was not tested but rather used in tests as an helper

Reply:
Can we add tests for all passes then?

Reply:
Aren't tests presented in https://github.com/openvinotoolkit/nncf/pull/1598, https://github.com/openvinotoolkit/nncf/pull/1642 cover enough?

Reply:
There were no tests for passes at all since they were used in the default pipeline.

Reply:
AFAIK, passes `remove_shapeof_subgraphs` and `filter_constant_nodes` were checked implicitly by graph tests. For my pass it was easier to create a synthetic model to check the pass